### PR TITLE
Support header-only file loading

### PR DESCRIPTION
**Pull request by OTDE**
This PR lets a user open a version of their EDF file with only header information.
Documentation will be adjusted in further commits to reflect this.

Performance is not affected by these changes:

previous:
```
julia> @benchmark EDF.read("test/data/test.edf")
BenchmarkTools.Trial: 
  memory estimate:  3.33 MiB
  allocs estimate:  6819
  --------------
  minimum time:     884.232 μs (0.00% GC)
  median time:      913.414 μs (0.00% GC)
  mean time:        1.014 ms (8.20% GC)
  maximum time:     2.654 ms (54.53% GC)
  --------------
  samples:          4924
  evals/sample:     1
```

current (this PR):
```
julia> @benchmark EDF.read("test/data/test.edf")
BenchmarkTools.Trial: 
  memory estimate:  3.32 MiB
  allocs estimate:  6814
  --------------
  minimum time:     863.867 μs (0.00% GC)
  median time:      893.672 μs (0.00% GC)
  mean time:        1.003 ms (8.20% GC)
  maximum time:     3.108 ms (41.48% GC)
  --------------
  samples:          4976
  evals/sample:     1

```

Solves a couple issues: #25 and #9

### CODE REVIEW

File: src/EDF.jl [hunk 1]
```diff
@@ -132,7 +132,7 @@ struct FileHeader
 end
 
 """
-    EDF.SignalHeader
+    EDF.Signal
```
Comment (jrevels) [inline src/EDF.jl:135]: I actually prefer `SignalHeader`; why the change back?

File: src/read.jl [hunk 2]
```diff
@@ -186,12 +202,35 @@ end
...
+function read!(file::File)
+    (isopen(file.io) || !eof(file.io)) && read_signals!(file)
+    return file
+end
+
+"""
+    EDF.open(path::AbstractString)
+
+Read the file, signal, and annotation-wide file headers from the EDF file at
+`path`, returning an `File` instance without reading the file's sample data.
```
Comment (jrevels) [inline src/read.jl:226]:
```suggestion
`path`, returning a `File` instance without reading the file's sample data.
```

File: src/EDF.jl (hunk 1 above)
Comment (OTDE) [reply src/EDF.jl:135]: My thinking was that this would make transitioning from the previous tagged versions a little smoother by keeping the same struct name, but `SignalHeader` is definitely more descriptive, and it may be to our benefit in this case to have the breaking changes be loud so nothing fails silently; I can totally switch it back.

File: src/read.jl (hunk 2 above)
Comment (OTDE) [reply src/read.jl:226]: ah, the old "used to be `EDF.File`" doc error. fixing!

File: src/EDF.jl (hunk 1 above)
Comment (jrevels) [reply src/EDF.jl:135]: Yeah, I'd say keep it. It's fine to break things if we bump the version number correctly. Anyway, I think `SignalHeader` makes even MORE sense given this particular change.

File: src/read.jl [hunk 3]
```diff
@@ -186,12 +202,35 @@ end
...
+The file's IO source remains open until closed manually with `close(file)`
+or automatically when calling `read!(file)`.
+"""
+function open(path::AbstractString)
```
Comment (ararslan) [inline src/read.jl:230]: We might additionally want to provide the same kind of API as `Base.open` which executes a function that will automatically close the file when the function finishes, e.g.
```julia
function open(f::Function, path::AbstractString)
    file = open(path)
    try
        return f(file)
    finally
        close(file)
    end
end
```

File: src/EDF.jl (hunk 1 above)
Comment (OTDE) [reply src/EDF.jl]: Fixed in https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/6163c31f3a4cb42399643f92820edb8950b670d7

File: src/read.jl (hunk 2 above)
Comment (OTDE) [reply src/read.jl]: Fixed in https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/6163c31f3a4cb42399643f92820edb8950b670d7

File: src/read.jl (hunk 3 above)
Comment (OTDE) [reply src/read.jl:228]: Fixed in https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/953dbd5d686371898c3e94d7ee1441825ad4fa50

File: src/EDF.jl [hunk 4]
```diff
@@ -168,39 +168,20 @@ Type representing the header record for an `AnnotationList`
...
+* `offset_in_file` (`Int`): The annotation header's position,
+   relative to other signals in its origin file
 """
 struct AnnotationListHeader
     n_samples::Int16
+    offset_in_file::Int
 end
 
-AnnotationListHeader(header::SignalHeader) = AnnotationListHeader(header.n_samples)
+AnnotationListHeader(header::SignalHeader, offset::Int) = AnnotationListHeader(header.n_samples, offset)
```
Comment (ararslan) [inline src/EDF.jl:179]:
```suggestion
function AnnotationListHeader(header::SignalHeader, offset::Integer)
    return AnnotationListHeader(header.n_samples, Int(offset))
end
```

File: .travis.yml [hunk 5]
```diff
@@ -1,23 +1,16 @@
 language: julia
 os:
   - linux
+dist: xenial
```
Comment (ararslan) [inline .travis.yml:4]: Why add this? Personally I think it's fine to let Travis use whatever its default Linux distribution is (which is currently Xenial).

File: .travis.yml [hunk 6]
```diff
@@ -1,23 +1,16 @@
...
-jobs:
-  include:
-    - stage: "Documentation"
-      julia: 1.0
-      os: linux
-      script:
-        - julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=".")); Pkg.instantiate()'
-        - julia --project=docs/ docs/make.jl
-      after_success: skip
+  - docs/build.sh
```
Comment (ararslan) [inline .travis.yml:16]: FWIW I think we only do this for private packages because there's currently no automated documentation deployment, so viewing rendered documentation has to be done locally. IMO we can keep it as-is for public packages.

File: .travis.yml (hunk 5 above)
Comment (OTDE) [reply .travis.yml:4]: For sure! I saw it being used in other packages in the beacosystem and thought there may be a consistency reason for it (I'm basing most CI choices here off OndaEDF, which may not be the wave). I'll revert if need be!

File: src/EDF.jl (hunk 4 above)
Comment (OTDE) [reply src/EDF.jl]: Fixed in https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/8e29613fe68cce4da7eaf83ef90ca6b10514ed97

File: .travis.yml (hunk 6 above)
Comment (OTDE) [reply .travis.yml]: Reverted here: https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/d51965b6ea43729cbb69d00e32ab9bc04dd8ded7

File: .travis.yml (hunk 5 above)
Comment (OTDE) [reply .travis.yml]: Reverted here: https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/d51965b6ea43729cbb69d00e32ab9bc04dd8ded7

**Comment by OTDE** [comment]
Thanks for y'all's patience with my CI inexperience! Let me know if anything else needs adjusting.

File: docs/Project.toml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 [deps]
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+EDF = "ccffbfc1-f56e-50fb-a33b-53d1781b2825"
```
Comment (ararslan) [inline docs/Project.toml:3]:
```suggestion
```
The documentation step in CI ensures that the right version of EDF.jl is added to the environment. If you have it listed as a project dependency and you inadvertently `instantiate` before you `develop`, you can end up in a weird state where the released version of EDF.jl is used instead of the checked out one, IIRC. Maybe that's not entirely accurate, I don't fully recall, but the general practice is to not include it here.

Comment (OTDE) [reply docs/Project.toml:3]: Sweet, that makes sense!

**Comment by OTDE** [comment]
Feedback addressed in https://github.com/beacon-biosignals/EDF.jl/pull/26/commits/9f33069b19b7deb7fdd1da4e8fc92eafb4962f6c !
